packages/components/src/components/tabs/tabs.jsx

```diff
@@ -1,141 +1,119 @@
-import classNames from 'classnames';
 import React from 'react';
+import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import Tab from './tab.jsx';
 
-class Tabs extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = { active_index: props.active_index || 0 };
-    }
-
-    onTabItemClick = index => {
-        this.setState({ active_index: index });
-    };
-
-    componentDidUpdate(prev_props, prev_state) {
-        if (this.props.active_index !== -1 && prev_state.active_index !== this.state.active_index) {
-            this.setState({ active_index: this.state.active_index || 0 });
-
-            if (typeof this.props.onTabItemClick === 'function') {
-                this.props.onTabItemClick(this.state.active_index);
-            }
-        }
-        if (this.props.active_index !== prev_props.active_index) {
-            this.setState({ active_index: this.props.active_index });
-        }
-        if (this.props.should_delay_render !== prev_props.should_delay_render && this.props.should_delay_render) {
-            this.setActiveLineStyle();
-        }
-    }
+const Tabs = ({
+    active_index,
+    bottom,
+    center,
+    children,
+    className,
+    fit_content,
+    header_fit_content,
+    onTabItemClick,
+    should_delay_render,
+    single_tab_has_no_label,
+    top,
+}) => {
+    const [active_line_style, setActiveLineStyle] = React.useState({});
+    const active_tab_ref = React.useRef(null);
+    const tabs_wrapper_ref = React.useRef(null);
+    const tab_width = fit_content ? '150px' : `${(100 / children.length).toFixed(2)}%`;
 
-    setActiveLineStyle = () => {
-        if (this.props.header_fit_content && this.active_tab_ref && this.tabs_wrapper_ref) {
-            const tabs_wrapper_bounds = this.tabs_wrapper_ref.getBoundingClientRect();
-            const active_tab_bounds = this.active_tab_ref.getBoundingClientRect();
-            if (active_tab_bounds.width === 0) {
-                setTimeout(() => {
-                    this.setActiveLineStyle();
-                }, 500);
```

Comment:
I believe we need this exactly as it's here. The reason is at the very first rendering phase, elements are not on the screen and their widths can't be calculated accordingly. So we need this if/else and the timeout.